crates/taplo/src/dom/node/nodes.rs

```diff
@@ -19,7 +19,7 @@ macro_rules! wrap_node {
     ) => {
         $(#[$attrs])*
         $vis struct $name {
-            pub(crate) inner: Arc<$inner>,
+            pub(crate) inner: Rc<$inner>,
```

Comment:
Why not use `Arc`?

Reply:
`Arc` is thread-safe, if threads are not a concern, `Rc` is better.

Reply:
Yeah, was wondering what the intention here was since I'm new to the codebase. It looked like `Arc` implementation can be replaced with `Rc` here as we don't enforce `Send` in the inner types. If these are breaking changes, I can revert these type changes as well as the `let else` changes which were stabilised in 1.65 rust.

Reply:
I would keep `Arc` to preserve usage in multi-threaded environment (there is some in the LSP and CLI as far as I remember). The benefit of `Rc` is useless because Taplo does not need performance. The project needs to be easily maintainable, safe, and correct before all.

I would also not use `let else` just yet. It's also not a huge benefit (actually I'm not even convinced it improves readability at all).

From a general point of view, Taplo is only in maintenance mode, so only critical changes are justified. However, if you have time to invest in the project, then I think it would make sense to make you a maintainer and you would then have more freedom to address larger code changes.

Reply:
Sounds good, got sidetracked but originally came for the issue which was resolved in #474